clients/src/main/java/org/apache/kafka/common/record/KafkaLZ4BlockOutputStream.java

```diff
@@ -259,10 +259,10 @@ private void ensureNotFinished() {
     @Override
     public void close() throws IOException {
         if (!finished) {
+            // basicallly flush the buffer writing the last block
             writeBlock();
+            // write the end block and finish the stream
             writeEndMark();
-            flush();
```

Comment:
@ijuma Yeah it `out` is properly implemented, then it should call `flush()` in its `close()`. But I am not sure Kafka should rely on this assumption since Java doc of `OutputStream.close()` does not explicitly say this behavior should be enforced. Also, following the same logic that a properly implemented system should call `flush()` inside `close()`, shouldn't Kafka do this as well?